test/funder.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * @typedef {Object} FundingReceipts
+ *
+ * @property {Array<Object>} originFaucetReceipts Array of receipts for
+ *                                                origin chain addresses funded
+ *                                                from mosaic faucet.
+ * @property {Array<Object>} auxiliaryFaucetReceipts Array of receipts for
+ *                                                   auxiliary chain addresses
+ *                                                   funded from mosaic faucet.
+ * @property {Array<Object>} ropstenFaucetReceipts Array of receipts for
+ *                                                 origin chain addresses funded
+ *                                                 from ropsten faucet
+ */
+
+/**
+ * @typedef {Object} FundingTransactionHashes
+ *
+ * @property {Array<string>} originFaucetTXHashes Array of transaction
+ *                                                hashes for origin chain
+ *                                                addresses funded from
+ *                                                mosaic faucet.
+ * @property {Array<string>} auxiliaryFaucetTXHashes Array of transaction
+ *                                                   hashes for auxiliary
+ *                                                   chain addresses funded
+ *                                                   from mosaic faucet.
+ * @property {Array<string>} ropstenFaucetTXHashes Array of transaction hashes
+ *                                                 for auxiliary for origin
+ *                                                 chain addresses funded
+ *                                                 from ropsten faucet
+ */
+
+
+/**
+ * @typedef {Object} FundingDetails
+ *
+ * @property {FundingReceipts} Receipts of all the faucet.
+ * @property {FundingTransactionHashes} transactionHashes of all the faucet.
+ */
+
+/**
+ * @typedef {Object} ERC20RefundTransaction
+ * @property {string}  type Type of fund.
+ * @property {string} faucetAddress Address of faucet.
+ * @property {string} tokenAddress Address of ERC20 contract.
+ */
+
+/**
+ * @typedef {Object} BaseTokenRefundTransaction
```

Comment:
If I recall correctly, we decided to call it `BaseCoin` to better differentiate from tokens.